src/lib/pages/tx-details/components/tx-message/msg-receipts/CoinsComponent.tsx

```diff
@@ -1,35 +1,34 @@
 import { Flex, Grid } from "@chakra-ui/react";
 import type { Coin } from "@cosmjs/stargate";
-import { isUndefined } from "lodash";
 import { useState } from "react";
 
 import { trackUseExpand } from "lib/amplitude";
 import { useMobile } from "lib/app-provider";
 import { ShowMoreButton } from "lib/components/button";
 import { UnsupportedTokensModal } from "lib/components/modal/UnsupportedTokensModal";
 import { TokenCard } from "lib/components/token/TokenCard";
-import type { AssetInfo, Option } from "lib/types";
+import { useAssetInfos } from "lib/services/assetService";
+import { useMovePoolInfos } from "lib/services/move";
 import { coinToTokenWithValue, filterSupportedTokens } from "lib/utils";
 
-type AssetObject = { [key: string]: AssetInfo };
-
-interface CoinComponentProps<
-  T extends Coin | Coin[],
-  A extends Option<AssetObject> | AssetObject,
-> {
-  amount: T;
-  assetInfos: A;
+interface CoinsComponentProps {
+  coins: Coin[];
 }
 
-const MultiCoin = ({
-  amount,
-  assetInfos,
-}: CoinComponentProps<Coin[], AssetObject>) => {
+export const CoinsComponent = ({ coins }: CoinsComponentProps) => {
   const isMobile = useMobile();
+  const { data: assetInfos } = useAssetInfos({
+    withPrices: true,
+  });
+  const { data: movePoolInfos } = useMovePoolInfos({
+    withPrices: true,
+  });
   const [showMore, setShowMore] = useState(false);
 
-  const tokens = amount.map((coin) =>
-    coinToTokenWithValue(coin.denom, coin.amount, assetInfos)
+  if (!assetInfos || !coins.length) return <>{JSON.stringify(coins)}</>;
```

Comment:
separate into 2 conditions.
1. !coin.length => "No assets"
2. !assetInfos => return the same.